testsuites/CBLTester/CBL_Functional_tests/TestSetup_FunctionalTests/test_replication.py

```diff
@@ -3215,6 +3215,260 @@ def test_replication_push_replication_guest_enabled(params_from_base_test_setup,
+    cookie, session_id = sg_client.create_session(sg_admin_url, sg_db, username)
+    session = cookie, session_id
+
+    # 3. push_pull replicate to SGW
+    replicator = Replication(base_url)
+    authenticator = Authenticator(base_url)
+    replicator_authenticator = authenticator.authentication(session_id, cookie, authentication_type="session")
+    repl = replicator.configure_and_replicate(source_db=cbl_db,
+                                              target_url=sg_blip_url,
+                                              continuous=True,
+                                              replicator_authenticator=replicator_authenticator)
+    # replicator.stop(repl)
+
+    # 4. remove doc A from channel A
+    doc_obj_A = db.getDocument(cbl_db, cbl_ids[0])
+    doc_A_mut = documentObj.toMutable(doc_obj_A)
+    doc_body_A = documentObj.toMap(doc_A_mut)
+    doc_body_A["channels"] = ["DEF"]
+    db.updateDocument(database=cbl_db, data=doc_body_A, doc_id=cbl_ids[0])
+
+    # 5. Remove doc B from channel A , B
+    doc_obj_B = db.getDocument(cbl_db, cbl_ids[1])
+    doc_B_mut = documentObj.toMutable(doc_obj_B)
+    doc_body_B = documentObj.toMap(doc_B_mut)
+    doc_body_B["channels"] = []
+    db.updateDocument(database=cbl_db, data=doc_body_B, doc_id=cbl_ids[1])
+
+    # 6. continue push_pull replication
+    replicator.wait_until_replicator_idle(repl)
+    replicator.stop(repl)
+
+    # 7. Verify user can only access doc A, but not doc B
+    sg_docs = sg_client.get_all_docs(url=sg_url, db=sg_db, include_docs=True, auth=session)["rows"]
+    assert len(sg_docs) == 1, "did not remove channels appropriately"
+    sg_doc_ids = [doc['id'] for doc in sg_docs]
+    assert cbl_ids[0] in sg_doc_ids, "doc A does not exist for the user"
+    assert cbl_ids[1] not in sg_doc_ids, "doc B exist for the user"
+
+
+@pytest.mark.listener
+@pytest.mark.replication
+def test_doc_removal_with_multipleChannels(params_from_base_test_setup, setup_customized_teardown_test):
+    """
+        @summary:
+        1. Create users in SGW with multiple channels
+            user A -> channelA,channelB, channel C;
+            userB -> channelB,
+            userC-> channelC
+        2. create docs in SGW
+            doc a with channel A, channelB ;
+            docb with channel B ,
+            docc with ChannelA, channelB, channelC
+        3. Verify User A can access docA and docC.
+            docB by UserB, UserA
+            docC by user A, user C
+        4. Remove the channel c from all the docs.
+        5. Verify userA can access only docA and doc B, but not docC
+            UserB can access docB
+            UserC cannot access docC
+    """
+
+    sg_db = "db"
+    sg_url = params_from_base_test_setup["sg_url"]
+    sg_admin_url = params_from_base_test_setup["sg_admin_url"]
+    sg_blip_url = params_from_base_test_setup["target_url"]
+    base_url = params_from_base_test_setup["base_url"]
+    cluster_config = params_from_base_test_setup["cluster_config"]
```

Comment:
duplicate variable declaration of "cluster_config" and "sg_config"

Reply:
done